src/network/serverpackethandler.cpp

```diff
@@ -1109,10 +1109,13 @@ void Server::handleCommand_Interact(NetworkPacket* pkt)
 			enable_anticheat && !isSingleplayer()) {
 		float d = player_pos.getDistanceFrom(pointed_pos_under);
 		if (!checkInteractDistance(player, d, pointed.dump())) {
-			// Re-send block to revert change on client-side
-			RemoteClient *client = getClient(pkt->getPeerId());
-			v3s16 blockpos = getNodeBlockPos(floatToInt(pointed_pos_under, BS));
-			client->SetBlockNotSent(blockpos);
+			if (action == 2 || action == 3) {
+				// Re-send the whole block to undo placement or dig prediction
+				RemoteClient *client = getClient(peer_id);
+				v3s16 blockpos = getNodeBlockPos(
```

Comment:
inline with next function call

Reply:
This is also one of the three points where I'm not sure if a mapblock or anything should be sent: https://github.com/minetest/minetest/pull/7630#issuecomment-412029621